packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -3128,19 +3110,107 @@ export class StripeHelper extends StripeHelperBase {
 
     const productPaymentCycleOld = this.stripePlanToPaymentCycle(planOld);
 
+    const sameBillingCycle =
+      productPaymentCycleOld === baseDetails.productPaymentCycleNew;
+
+    // get next invoice details
+    const nextInvoice = upcomingInvoiceWithInvoiceItem
+      ? upcomingInvoiceWithInvoiceItem
+      : await this.previewInvoiceBySubscriptionId({
+          subscriptionId: subscription.id,
+        });
+
+    const {
+      total: nextInvoiceTotal,
+      currency: nextInvoiceCurrency,
+      lines: nextInvoiceLines,
+    } = nextInvoice || {};
+
+    // https://stripe.com/docs/api/invoices/object#invoice_object-lines
+    // as per the Stripe docs, one of the individual line items includes
+    // new plan pricing and tax information without additional fees
+    const upcomingInvoiceLineItem =
+      nextInvoiceLines && nextInvoiceLines.data
+        ? nextInvoiceLines.data.find((line) => !line.proration)
+        : undefined;
+
+    const {
+      amount: upcomingInvoiceAmount,
+      discount_amounts: upcomingInvoiceDiscountAmounts,
+      tax_amounts: upcomingInvoiceTaxAmounts,
+    } = upcomingInvoiceLineItem || {};
+
+    const sameBillingCondition =
+      sameBillingCycle && upcomingInvoiceLineItem !== undefined;
+
+    const newListPriceTotalWithoutTax =
+      sameBillingCondition && upcomingInvoiceAmount !== undefined
+        ? upcomingInvoiceAmount
+        : undefined;
```

Comment:
This could be refactored to the following.

```
const newListPriceTotalWithoutTax = sameBillingCondition && upcomingInvoiceAmount;
```

Reply:
I originally condensed it to this but it went from `newListPriceTotalWithoutTax` being `number | undefined` (what I want) to `number | false | undefined`, which ended up giving wrong numbers for one of the aforementioned test cases. I can't remember off the top of my head which one.